airbyte-integrations/connectors/source-mailchimp/source_mailchimp/streams.py

```diff
@@ -0,0 +1,178 @@
+        decoded_response = response.json()
+        api_data = decoded_response[self.data_field]
+        if len(api_data) < self.PAGINATION:
+            self.current_offset = 0
+            return {}
+        else:
+            self.current_offset += self.PAGINATION
+            return {"offset": self.current_offset}
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+
+        params = {"count": self.PAGINATION}
+
+        # Handle pagination by inserting the next page's token in the request parameters
+        if next_page_token:
+            params.update(next_page_token)
+        return params
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        yield from response_json[self.data_field]
+
+    @property
+    @abstractmethod
+    def data_field(self) -> str:
+        """the responce entry that contains useful data"""
+        pass
+
+
+class IncrementalMailChimpStream(MailChimpStream, ABC):
+    state_checkpoint_interval = math.inf
+
+    @property
+    @abstractmethod
+    def cursor_field(self) -> str:
+        """
+        Defining a cursor field indicates that a stream is incremental, so any incremental stream must extend this class
+        and define a cursor field.
+        """
+        pass
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+        latest_state = latest_record.get(self.cursor_field)
+        current_state = current_stream_state.get(self.cursor_field) or latest_state
+        return {self.cursor_field: max(latest_state, current_state)}
+
+    def request_params(self, stream_state=None, **kwargs):
+
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        default_params = {"sort_field": self.cursor_field, "sort_dir": "ASC"}
+        since_value = stream_state.get(self.cursor_field)
+        if since_value:
+            default_params[f"since_{self.cursor_field}"] = since_value
+        params.update(default_params)
+        return params
+
+
+class Lists(IncrementalMailChimpStream):
+    cursor_field = "date_created"
+    data_field = "lists"
+
+    def path(self, **kwargs) -> str:
+        return "lists"
+
+
+class Campaigns(IncrementalMailChimpStream):
+    cursor_field = "create_time"
+    data_field = "campaigns"
+
+    def path(self, **kwargs) -> str:
+        return "campaigns"
+
+
+class EmailActivity(IncrementalMailChimpStream):
+    cursor_field = "timestamp"
+    name = "Email_activity"
```

Comment:
don't need to have custom name

Reply:
just have all schema files in lower case please, lets be consistent with all CDK connectors, this will simplify reading